docs/testing/unit-testing.md

```diff
@@ -0,0 +1,413 @@
+          this.paragraph.text = paragraphText;
+          this.paragraph.insertLocation = insertLocation;
+          return this.paragraph;
+        },
+      },
+    },
+  },
+  // Mock the Word.InsertLocation enum.
+  InsertLocation: {
+    end: "end",
+  },
+  // Mock the Word.run method.
+  run: async function(callback) {
+    await callback(this.context);
+  },
+};
+
+// Create the final mock object from the seed object.
+const wordMock = new OfficeAddinMock.OfficeMockObject(mockData);
+
+// Define and initialize the Word object that is called in the insertBlueParagraph function.
+global.Word = wordMock;
+
+/* Code that calls the test framework goes below this line. */
+
+// Jest test set
+describe("Insert blue paragraph at end tests", () => {
+
+  test("color of paragraph", async function () {
+    await myWordAddinFeature.insertBlueParagraph();  
+    expect(wordMock.context.document.body.paragraph.font.color).toBe("blue");
+  });
+
+  test("text of paragraph", async function () {
+    await myWordAddinFeature.insertBlueParagraph();
+    expect(wordMock.context.document.body.paragraph.text).toBe("Hello World");
+  });
+})
+```
+
+## Adding mock objects, properties, and methods dynamically when testing
+
+In some scenarios, efficient testing requires that mock objects be created or modified at runtime; that is, while the tests are running. The following are examples:
+
+- The function that is being tested behaves differently when called a second time, from how it behaves when first called. You need to first test the function with one mock object, then change this mock object and test the function again with the changed mock object.
+- You need to test a function against multiple arguments quickly. For example, you need to test a function with a mock object that has a color property and then test the function again with a mock object that has a text property, but is otherwise identical to the original mock object.
+
+The `OfficeMockObject` has three methods to help in these scenarios.
+
+- `OfficeMockObject.setMock` adds a property and value to a `OfficeMockObject` object. The following is an example:
+
+    ```javascript
+    rangeMock.setMock("address", "G6:K9");
+    ```
+
+- `OfficeMockObject.addMockFunction` adds a mock function to a `OfficeMockObject` object. If the function parameter isn't used, an empty function is created. The following is an example:
+
+    ```javascript
+    workbookMock.addMockFunction("getSelectedRange", function () { 
+      const range = {
+        address: "B2:G5",
+      };
+      return range;
+    });
+    ```
+
+- `OfficeMockObject.addMock` adds a new `OfficeMockObject` object as a property to an existing one and gives it a name. It would have the minimum members that all `OfficeMockObject` have, such as `load` and `sync`. Additional members can be added with the `setMock` and `addMockFunction` methods. The following is an example:
+
+    ```javascript
+    workbookMock.addMock("range");
```

Comment:
Is this adding a mock property called "range" to the mock workbook? I'm confused because this doesn't map to my mental picture of the Excel object model.